### ISSUE TITLE
Do you know how to change the rotation axis?

### ISSUE THREAD

**Issue by likecanyon** [leap-hand/LEAP_Hand_Sim#26]
Hello Kenny, do you know how to change the rotation axis to make the hand can rotate the cube around other axis such as y axis.
I modified the self.cfg["env"]["rotation_axis"] to [0.,1.,0.] in leap_hand_rot.py and define  self.rot_axis_buf[:, 0] = 0  # Zero out x-axis rotation
        self.rot_axis_buf[:, 1] = 1  # Only y-axis rotation is rewarded 
        self.rot_axis_buf[:, 2] = 0  # Zero out z-axis rotation  
        in the compute_reward method in leap_hand_rot.py, but the new  tained policy based on the modification showed nothing difference, the cube was still rotated around z axis.
        Then I added 1 rotation reward about y axis and penalties around x,z axes and added them to the original rewards in the compute_reward, however, after training, the agent still didn't rotate the cube around y axis, could you please offer any suggestion?  
         def compute_reward_align_y_axis(self, object_angvel, rotate_reward_scale=2.0):
        # Define the y-axis unit vector
        y_axis = torch.tensor([0.0, 1.0, 0.0], device=object_angvel.device).unsqueeze(0).repeat(object_angvel.shape[0], 1)
        
        # Normalize angular velocity
        angvel_norm = torch.norm(object_angvel, dim=1, keepdim=True) + 1e-6  # Prevent division by zero
        angvel_unit = object_angvel / angvel_norm

        # Calculate the alignment (dot product)
        alignment = torch.sum(angvel_unit * y_axis, dim=1)

        # Clip the alignment between -1 and 1
        alignment = torch.clamp(alignment, min=-1.0, max=1.0)

        # Scale the reward
        rotate_reward_y = rotate_reward_scale * alignment

        return rotate_reward_y

    def compute_reward_penalize_other_axes(self, object_angvel, penalty_scale=10.0):
        # Define the x and z-axis unit vectors
        x_axis = torch.tensor([1.0, 0.0, 0.0], device=object_angvel.device).unsqueeze(0).repeat(object_angvel.shape[0], 1)
        z_axis = torch.tensor([0.0, 0.0, 1.0], device=object_angvel.device).unsqueeze(0).repeat(object_angvel.shape[0], 1)

        # Calculate penalties as the absolute angular velocities along x and z
        penalty_x = torch.abs(torch.sum(object_angvel * x_axis, dim=1))
        penalty_z = torch.abs(torch.sum(object_angvel * z_axis, dim=1))

        # Total penalty
        total_penalty_xz = penalty_scale * (penalty_x + penalty_z)

        return -total_penalty_xz  # Negative to penalize

**Comment by Lucky7Chess** [comment]
You may need to add cube pose to the policy observation space, not sure if it'll train on a new axis as is.  It's quite a bit harder to get that axis to work without pose though it should be possible.

**Comment by likecanyon** [comment]
Thank you for your prompt reply and valuable suggestion, I will check it out. Best luck of your ongoing research : D

**Comment by lixiaolong04** [comment]
> 感谢您的及时回复和宝贵建议，我会检查一下。祝您正在进行的研究好运 ： D

Hi, I'm also working on rotating objects around other axes by hand. I tried to modify the rotation axis training, the real effect of rotation is not ideal. Do you have any new ideas? Thank you very much for your reply and suggestions.

**Comment by Lucky7Chess** [comment]
You may also want to check the Mujoco playground example, they have 6 axis rotation working there with the leap hand.

**Comment by lixiaolong04** [comment]
> 您可能还想查看 Mujoco playground 示例，它们有 6 轴旋转与跳跃指针一起工作。

Ok, thank you very much for your reply

**Comment by lixiaolong04** [comment]
> You may also want to check the Mujoco playground example, they have 6 axis rotation working there with the leap hand.

Hello, I'm very interested in this Mujoco Playground. I have already completed the reproduction in the Leaphand simulation environment. Now I want to do sim2real, but this part of the code is not open source. Do you have any good suggestions? Thank you for your reply.

**Comment by Lucky7Chess** [comment]
You may want to contact MuJoCo playground about that, I am not an author of that
